File: c#/Consumer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace LB1
{
    class Consumer
    {
        private Manager manager;
        private int idConsumer;
        private int countProduct;

        public Consumer(Manager manager, int countProduct, int idConsumer)
        {
            this.manager = manager;
            this.idConsumer = idConsumer;
            this.countProduct = countProduct;
        }
        public void Run()
        {
            for (int i = 0; i < countProduct; i++)
            {
                Thread.Sleep(1000);

                manager.fullStock.WaitOne();
                manager.takeItem.WaitOne();

                if (manager.StockSize > 0)
                {
                    string product = manager.ItemStock;
                    Console.WriteLine($"Consumer {idConsumer} taken from stock {product}");

                    manager.takeItem.Release();
                    manager.emptyStock.Release();
                }
                else
                {
                    manager.takeItem.Release();
                    break;
                }
            }
            Console.WriteLine("Consumer " + idConsumer + " has finished work");
        }
    }
}

File: java/Consumer.java
public class Consumer extends Thread {
    private final int idConsumer;
    private Manager manager;
    private int countProduct;

    public Consumer(Manager manager,int countProduct, int idConsumer) {
        this.idConsumer = idConsumer;
        this.manager = manager;
        this.countProduct = countProduct;
    }

    @Override
    public void run() {
        for (int i = 0; i < countProduct; i++)
        {
            try {
                Thread.sleep(2000);
                manager.emptyStock.acquire();
                manager.takeItem.acquire();

                if (manager.isProducingWorkDone && manager.getStockSize() == 0) {
                    manager.takeItem.release();
                    break;
                }

                if (manager.getStockSize() > 0 && manager.generalCountProductUsing > 0) {
                    String item = manager.getItemStock();
                    System.out.println("Consumer " + idConsumer + " taken from stock " + item);
                }

                manager.takeItem.release();
                manager.fullStock.release();

            } catch (InterruptedException e) {
                throw new RuntimeException(e);
            }
        }
        System.out.println("Consumer " + idConsumer + " has finished work");
    }
}
